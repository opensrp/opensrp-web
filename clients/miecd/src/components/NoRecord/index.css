.norecord{
    text-align: center;
    border: 1px solid #dadce0;
    background-color: #dadce0;
    width: 25rem;
    margin: auto;
    color: #3a484f;
}