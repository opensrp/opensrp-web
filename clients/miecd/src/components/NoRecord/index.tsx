import React from 'react';
import './index.css';

/**
 * ineterface for NoRecord object props
 * @member {string} message - message to be showed by the component
 */
interface Props {
    message: string;
}

/**
 * The NoRecord component shows a message passed to it
 * as a prop to the user
 * @param {string} message - message to be showed by the component
 */
export default function NoRecord({ message = '' }: Props) {
    return <div className="norecord">{message}</div>;
}
