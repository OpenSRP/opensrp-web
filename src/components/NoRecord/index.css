.norecord{
    text-align: center;
    border: 1px solid #dadce0;
    background-color: #dadce0;
    width: 20rem;
    margin: auto;
}