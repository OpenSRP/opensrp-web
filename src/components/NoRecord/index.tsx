import React from 'react';

export default function NoRecord() {
  return <div>no record</div>;
}   
