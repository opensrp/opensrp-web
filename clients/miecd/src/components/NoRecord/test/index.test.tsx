import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import NoRecord from '..';

describe('src/components/NoRecord', () => {
    it('must render without crashing', () => {
        shallow(<NoRecord message="test message" />);
    });
    it('must render correctly', () => {
        const wrapper = mount(<NoRecord message="test message" />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
