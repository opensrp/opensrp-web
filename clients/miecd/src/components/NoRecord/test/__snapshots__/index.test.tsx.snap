// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/NoRecord must render correctly 1`] = `
<NoRecord
  message="test message"
>
  <div
    className="norecord"
  >
    test message
  </div>
</NoRecord>
`;
